webapp/application.py

```diff
@@ -125,4 +131,178 @@ def application_page(token):
         application=application,
         job=job,
         hiring_lead=hiring_lead,
+        token=token,
+    )
+
+
+@application.route("/withdraw/<string:token>")
+def application_withdrawal(token):
+    payload = verification_token_cipher.decrypt(token)
+    try:
+        payload = json.loads(payload)
+        email = payload.get("email")
+        withdrawal_reason = payload.get("withdrawal_reason", "")
+        candidate_id = payload.get("candidate_id")
+        application_id = payload.get("application_id")
+    except (ValueError, TypeError):
+        flask.abort(404)
```

Comment:
This try catch block it too greedy. It's gonna be pretty misleading to mask all the potential errors here with a 404 page.

Why wouldn't you want to inform the user / developers about `ValueError`s and `TypeError`s anyway?